onnxruntime/test/providers/nnapi/nnapi_basic_test.cc

```diff
@@ -39,6 +42,27 @@ namespace test {
 
 #if !defined(ORT_MINIMAL_BUILD)
 
+using UnpackTensorFn = std::function<void(const InitializedTensorSet& initializers,
+                                          const std::string& name, std::vector<uint8_t>& unpacked_tensor)>;
+
+/* NNAPI QDQ model build test related helper functions */
+
+void GetQDQModelQuantizationScales(const Path& model_path, float scale,
+                                   const std::string& name, const InitializedTensorSet& initializers,
+                                   const UnpackTensorFn& unpack_tensor) {
+  std::vector<uint8_t> unpacked_tensor;
+  unpack_tensor(initializers, name, unpacked_tensor);
+  scale = reinterpret_cast<const float*>(unpacked_tensor.data())[0];
```

Comment:
>scale [](http://example.com/codeflow?start=2&length=5)

if using an output parameter, it should be `float& scale`. but consider just returning the single value instead.

also, instead of `reinterpret_cast<float*>` consider using memcpy instead. reinterpret_cast can be tricky to use safely.

[https://en.cppreference.com/w/cpp/language/reinterpret_cast](https://en.cppreference.com/w/cpp/language/reinterpret_cast)
from that link:
"When it is needed to interpret the bytes of an object as a value of a different type, std::memcpy or std::bit_cast (since C++20)can be used"